### PR TITLE
Ciena Devices

### PR DESCRIPTION
**Pull request by jgroom33**
Ready for review

### CODE REVIEW

**Comment by jgroom33** [comment]
@danner26 Do I need to do something else with this before merge? Would you prefer the commits are squashed?

**Comment by harryajc** [comment]
Hi @jgroom33 Thanks for your contribution

I am going to look at reviewing this but given the large amount of files changed this may take some time.

Harry

**Review by danner26 — APPROVED** [review]

**Review by harryajc — CHANGES_REQUESTED** [review]

File: device-types/Ciena/5171-platform.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    type: 10gbase-x-sfpp
+  - name: '37'
+    type: 10gbase-x-sfpp
+  - name: '38'
+    type: 10gbase-x-sfpp
+  - name: '39'
+    type: 10gbase-x-sfpp
+  - name: '40'
+    type: 10gbase-x-sfpp
+  - name: MGMT
```
[the remark this answers is not in the document]
Comment (jgroom33) [reply device-types/Ciena/5171-platform.yaml:115]: Thanks. Updated. Oddly enough, its ports 10,11,30,31

**Review by harryajc — APPROVED** [review]

**Comment by CADbloke** [comment]
FYI the pipe symbol `|` is illegal in Windows file names and will not check out.

````
C:\>cd Codez\Github\devicetype-library

C:\Codez\Github\devicetype-library>git restore --source=HEAD :/
error: invalid path 'module-types/Ciena/81(10|14)-12x25G-sfp28.yaml'
error: invalid path 'module-types/Ciena/81(10|14)-2x400G-cfp2.yaml'
error: invalid path 'module-types/Ciena/81(10|14)-2x400G-qsfpdd-macsec.yaml'
error: invalid path 'module-types/Ciena/81(10|14)-2x400G-qsfpdd.yaml'
error: invalid path 'module-types/Ciena/81(10|14)-8x100G-qsfp28.yaml'
error: invalid path 'module-types/Ciena/AC-39(26|30|32)|5142.yaml'
error: invalid path 'module-types/Ciena/AC-39(40|60).yaml'
error: invalid path 'module-types/Ciena/AC-51(62|70).yaml'
error: invalid path 'module-types/Ciena/AC-81(10|14).yaml'
error: invalid path 'module-types/Ciena/AC-81(40|90).yaml'
error: invalid path 'module-types/Ciena/DC-51(62|70).yaml'
error: invalid path 'module-types/Ciena/DC-81(10|14).yaml'
error: invalid path 'module-types/Ciena/DC-81(40|90).yaml'
````

**Comment by AskskwBv8T2nrm4Qnj** [comment]
I am also facing errors related to this commit, when trying to git pull my synced repository in my IntelliJ Idea:
`POST git-upload-pack (327 bytes) From https://github.com/AskskwBv8T2nrm4Qnj/devicetype-library * branch master -> FETCH_HEAD = [up to date] master -> origin/master invalid path 'module-types/Ciena/81(10|14)-12x25G-sfp28.yaml' invalid path 'module-types/Ciena/81(10|14)-2x400G-cfp2.yaml' invalid path 'module-types/Ciena/81(10|14)-2x400G-qsfpdd-macsec.yaml' invalid path 'module-types/Ciena/81(10|14)-2x400G-qsfpdd.yaml' invalid path 'module-types/Ciena/81(10|14)-8x100G-qsfp28.yaml' invalid path 'module-types/Ciena/AC-39(26|30|32)|5142.yaml' invalid path 'module-types/Ciena/AC-39(40|60).yaml' invalid path 'module-types/Ciena/AC-51(62|70).yaml' invalid path 'module-types/Ciena/AC-81(10|14).yaml' invalid path 'module-types/Ciena/AC-81(40|90).yaml' invalid path 'module-types/Ciena/DC-51(62|70).yaml' invalid path 'module-types/Ciena/DC-81(10|14).yaml' invalid path 'module-types/Ciena/DC-81(40|90).yaml'`

**Pull request merged**